### ISSUE TITLE
Is there any chance that instead of going with nginx and letencrypt, we have a Traefik for reverse proxy and ssl?

### ISSUE THREAD

**Issue by mehdiamenein** [caprover/caprover#947]
**Is your feature request related to a problem? Please describe.**
In some scenarios and occasions it can get tricky to modify the nginx config file and there is no UI structure for it however if we run traefik as a container then we can easier manage SSL and control other containers also with labels for reverse proxy and SSL with letencrypt

**Describe the solution you'd like**
Instead of nginx config files, containers have a section for labels to set up traefik per domain

**Describe alternatives you've considered**
here is a example of traefik in docker-compose that worked  for me:
```
version: "3"
services:
  traefik:
    image: traefik:v2.0
    restart: always
    container_name: traefik
    ports:
      - "80:80"
      - "443:443"
    command:
      - --api.insecure=false
      - --api.dashboard=false
      - --api.debug=false
      - --log.level=DEBUG
      - --providers.docker=true
      - --providers.docker.exposedbydefault=false
      - --providers.file.filename=/dynamic.yaml
      - --providers.docker.network=web
      - --entrypoints.web.address=:80
      - --entrypoints.web-secured.address=:443
      - --certificatesresolvers.mytlschallenge.acme.tlschallenge=true
      - --certificatesresolvers.mytlschallenge.acme.email=example@mail.com
      - --certificatesresolvers.mytlschallenge.acme.storage=/letsencrypt/acme.json
    volumes:
      - ./letsencrypt:/letsencrypt
      - /var/run/docker.sock:/var/run/docker.sock
      - ./dynamic.yaml:/dynamic.yaml
    networks:
      - web

  core:
    image: some-app-image
    command: bash -c "rm -f tmp/pids/server.pid && bundle exec rails s -p 3000 -b '0.0.0.0'"
    ports:
      - "3000:3000"
    networks:
      - web
    labels:
      #### Labels define the behavior and rules of the traefik proxy for this container ####
      - "traefik.enable=true" # <== Enable traefik to proxy this container
      - "traefik.http.routers.core-web.rule=Host(`sub.domain.com`)" # <== Your Domain Name goes here for the http rule
      - "traefik.http.routers.core-web.entrypoints=web" # <== Defining the entrypoint for http, **ref: line 30
      - "traefik.http.services.core-web.loadbalancer.server.port=3000"
      - "traefik.http.routers.core-web.middlewares=redirect@file" # <== This is a middleware to redirect to https
      - "traefik.http.routers.core-secured.rule=Host(`sub.domain.com`)" # <== Your Domain Name for the https rule
      - "traefik.http.routers.core-secured.entrypoints=web-secured" # <== Defining entrypoint for https, **ref: line 31
      - "traefik.http.routers.core-secured.tls.certresolver=mytlschallenge" # <== Defining certsresolvers for https

  front:
    image: some-app-image
    networks:
      - web
    ports:
      - "8080:8080"
    labels:
      #### Labels define the behavior and rules of the traefik proxy for this container ####
      - "traefik.enable=true" # <== Enable traefik to proxy this container
      - "traefik.http.routers.front-web.rule=Host(`sub2.domain.com`)" # <== Your Domain Name goes here for the http rule
      - "traefik.http.routers.front-web.entrypoints=web" # <== Defining the entrypoint for http, **ref: line 30
      - "traefik.http.services.front-web.loadbalancer.server.port=8080"
      - "traefik.http.routers.front-web.middlewares=redirect@file" # <== This is a middleware to redirect to https
      - "traefik.http.routers.front-secured.rule=Host(`sub2.domain.com`)" # <== Your Domain Name for the https rule
      - "traefik.http.routers.front-secured.entrypoints=web-secured" # <== Defining entrypoint for https, **ref: line 31
      - "traefik.http.routers.front-secured.tls.certresolver=mytlschallenge" # <== Defining certsresolvers for https

```

and also here is the dynamic.yaml file:

```
## Setting up the middleware for redirect to https ##
http:
  middlewares:
    redirect:
      redirectScheme:
        scheme: https
```

**Comment by githubsaturn** [comment]
Duplicate of https://github.com/caprover/caprover/issues/578#issuecomment-570935199

**Comment by githubsaturn** [comment]
TLDR: replacing nginx+let's encrypt with Traefik is essentially a whole new project and requires rewriting lots of bits and pieces of CapRover.
